src/nimony/sem.nim

```diff
@@ -3001,8 +3004,20 @@ proc semTypeSection(c: var SemContext; n: var Cursor) =
 
   c.addSym delayed
   wantParRi c, n
+
+
   publish c, delayed.s.name, declStart
 
+  if isEnumTypeDecl:
+    var dest = createTokenBuf()
+    var enumTypeDecl = cursorAt(c.dest, declStart)
+    genEnumToStrProc(dest, enumTypeDecl, c.types.stringType)
+    endRead(c.dest)
+    var dollorProcDecl = beginRead(dest)
+    var it = Item(n: dollorProcDecl, typ: c.types.autoType)
+    semExpr(c, it)
```

Comment:
A bit ugly as the thing is so simple that it shouldn't require semchecking. But alright for now.